services/notifications/pkg/email/email.go

```diff
@@ -13,7 +17,33 @@ var (
 )
 
 // RenderEmailTemplate renders the email template for a new share
-func RenderEmailTemplate(templateName string, templateVariables map[string]string, emailTemplatePath string) (string, error) {
+func RenderEmailTemplate(et MessageTemplate, locale string, emailTemplatePath string, translationPath string, vars map[string]interface{}) (string, string, error) {
+	rawsub := ComposeMessage(et.Subject, locale, translationPath)
+	// replace the body email placeholders with the values
+	subject, err := executeRaw(rawsub, vars)
+	if err != nil {
+		return "", "", err
+	}
+
+	bodyPlaceholders := map[string]interface{}{}
```

Comment:
Can we use a `MessageTemplate` instead a `map[string]interface{}`? Imo the fields will always match.